### PR TITLE
test: cross-check more tests against mlir-opt

### PR DESCRIPTION
**Pull request by tobiasgrosser**
Add MLIR_ROUNDTRIP to 6 tests and MLIR_UNREGISTERED_ROUNDTRIP to 27 that already pass them, and MLIR_INVALID to the 90 negative tests that mlir-opt also rejects. The Parsing tests embed error line numbers in their CHECK lines, which move by one with the extra RUN line.

### CODE REVIEW

File: Test/Verifier/arith_i0_operand.mlir [hunk 1]
```diff
@@ -1,4 +1,5 @@
 // RUN: not veir-opt %s 2>&1 | filecheck %s
+// RUN: %if mlir-min-23 %{ MLIR_INVALID %}
```
Comment (math-fehr) [inline Test/Verifier/arith_i0_operand.mlir:2]: Interesting, why did this require MLIR 23? Did something change before that?

File: Test/Verifier/arith_i0_result.mlir [hunk 2]
```diff
@@ -1,4 +1,5 @@
 // RUN: not veir-opt %s 2>&1 | filecheck %s
+// RUN: %if mlir-min-23 %{ MLIR_INVALID %}
```
Comment (math-fehr) [inline Test/Verifier/arith_i0_result.mlir:2]: Interesting, why did this require MLIR 23? Did something change before that?

**Review by math-fehr — APPROVED** [review]
Nice! I just had a question about two tests (for my own knowledge, nothing blocking). Feel free to merge after that.

File: Test/Verifier/arith_i0_operand.mlir (hunk 1 above)
Comment (tobiasgrosser) [reply Test/Verifier/arith_i0_operand.mlir:2]: MLIR stopped to support i0 integers in arith from 23, but still allowed them beforehand.
